src/rust/pal/data_structures.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//==============================================================================
+// Windows data structures
+//==============================================================================
```

Comment:
Same thing here, group the data structures by type, rather than platform.

Reply:
Agreed, Pedro and I had planned to do this in subsequent PR(s). [As per the next steps listed in the PR description above: (1) Refactor the PAL to make platform/subsystem specific hierarchy in the directory structure.]